util/runtime/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Tokio Core Reactor wrapper.
+
+extern crate futures;
+extern crate tokio;
+
+use std::{fmt, thread};
+use std::sync::mpsc;
+use std::time::{Duration, Instant};
+use futures::{future, Future, IntoFuture};
+pub use tokio::timer::Delay;
+pub use tokio::runtime::{Runtime as TokioRuntime, Builder as TokioRuntimeBuilder, TaskExecutor};
+
+/// Runtime for futures.
+///
+/// Runs in a separate thread.
+pub struct Runtime {
+	executor: Executor,
+	handle: RuntimeHandle,
+}
+
+impl Runtime {
+	fn new(runtime_bldr: &mut TokioRuntimeBuilder) -> Self {
+		let mut runtime = runtime_bldr
+			.build()
+			.expect("Building a Tokio runtime will only fail when mio components \
+				cannot be initialized (catastrophic)");
+		let (stop, stopped) = futures::oneshot();
+		let (tx, rx) = mpsc::channel();
+		let handle = thread::spawn(move || {
+
+			tx.send(runtime.executor()).expect("Rx is blocking upper thread.");
+			runtime.block_on(futures::empty().select(stopped).map(|_| ()).map_err(|_| ()))
+				.expect("Tokio runtime should not have unhandled errors.");
+		});
+		let executor = rx.recv().expect("tx is transfered to a newly spawned thread.");
+
+		Runtime {
+			executor: Executor {
+				inner: Mode::Tokio(executor),
+			},
+			handle: RuntimeHandle {
+				close: Some(stop),
+				handle: Some(handle),
+			},
+		}
+	}
+
+	/// Spawns a new tokio runtime with a default thread count on a background
+	/// thread and returns a `Runtime` which can be used to spawn tasks via
+	/// its executor.
+	pub fn with_default_thread_count() -> Self {
+		let mut runtime_bldr = TokioRuntimeBuilder::new();
+		Self::new(&mut runtime_bldr)
+	}
+
+	/// Spawns a new tokio runtime with a the specified thread count on a
+	/// background thread and returns a `Runtime` which can be used to spawn
+	/// tasks via its executor.
+	pub fn with_thread_count(thread_count: usize) -> Self {
+		let mut runtime_bldr = TokioRuntimeBuilder::new();
+		runtime_bldr.core_threads(thread_count);
+
+		Self::new(&mut runtime_bldr)
+	}
+
+	/// Returns this runtime raw executor.
+	///
+	/// Deprecated: Exists only to connect with current JSONRPC implementation.
```

Comment:
Maybe use deprecation attribute?
```suggestion
	#[deprecated(since = "0.1.0", note = "Exists only to connect with current JSONRPC implementation.")]

```